examples/pytorch/pytorch_checkpoint.py

```diff
@@ -0,0 +1,194 @@
+        model.train()
+        for batch_idx, (data, target) in enumerate(train_loader):
+            # Limiting training data for faster epochs.
+            if batch_idx * BATCHSIZE >= N_TRAIN_EXAMPLES:
+                break
+
+            data, target = data.view(data.size(0), -1).to(DEVICE), target.to(DEVICE)
+
+            optimizer.zero_grad()
+            output = model(data)
+            loss = F.nll_loss(output, target)
+            loss.backward()
+            optimizer.step()
+
+        # Validation of the model.
+        model.eval()
+        correct = 0
+        with torch.no_grad():
+            for batch_idx, (data, target) in enumerate(valid_loader):
+                # Limiting validation data.
+                if batch_idx * BATCHSIZE >= N_VALID_EXAMPLES:
+                    break
+                data, target = data.view(data.size(0), -1).to(DEVICE), target.to(DEVICE)
+                output = model(data)
+                # Get the index of the max log-probability.
+                pred = output.argmax(dim=1, keepdim=True)
+                correct += pred.eq(target.view_as(pred)).sum().item()
+
+        accuracy = correct / min(len(valid_loader.dataset), N_VALID_EXAMPLES)
+
+        trial.report(accuracy, epoch)
+
+        # Save optimization status.
+        torch.save(
+            {
+                "epoch": epoch,
+                "model_state_dict": model.state_dict(),
+                "optimizer_state_dict": optimizer.state_dict(),
+                "accuracy": accuracy,
+            },
+            os.path.join(path, "model.pt"),
+        )
+
+        # Handle pruning based on the intermediate value.
+        if trial.should_prune():
+            raise optuna.exceptions.TrialPruned()
+
+    return accuracy
+
+
+def restart_from_checkpoint(study, trial):
+    # Enqueue trial with the same parameters as the stale trial to use saved information.
+
+    path = f"pytorch_checkpoint/{trial.number}/model.pt"
+    if os.path.exists(path):
+        user_attrs = dict(**trial.user_attrs, **{"checkpoint_path": path})
+    else:
+        user_attrs = trial.user_attrs
+
+    study.add_trial(
+        optuna.create_trial(
+            state=optuna.trial.TrialState.WAITING,
+            params=trial.params,
+            distributions=trial.distributions,
+            user_attrs=user_attrs,
+            system_attrs=trial.system_attrs,
+        )
+    )
+
+
+if __name__ == "__main__":
+    storage = optuna.storages.RDBStorage(
+        "sqlite:///example.db", heartbeat_interval=1, failed_trial_callback=restart_from_checkpoint
+    )
+    study = optuna.create_study(
+        storage=storage, study_name="pytorch_checkpoint", direction="maximize", load_if_exists=True
+    )
+    study.optimize(objective, n_trials=100, timeout=600)
+
+    pruned_trials = [t for t in study.trials if t.state == optuna.trial.TrialState.PRUNED]
+    complete_trials = [t for t in study.trials if t.state == optuna.trial.TrialState.COMPLETE]
```

Comment:
Could you use [`Study.get_trials` with `states` argument](https://optuna.readthedocs.io/en/latest/reference/generated/optuna.study.Study.html?highlight=get_trials#optuna.study.Study.get_trials) instead?